src/client.ts

```diff
@@ -36,98 +44,52 @@ import { Logger, LoggerConfig, makeLogger, defaultLogger } from './logger/logger
  * })
  */
 export class SpvWalletClient {
-  serverUrl: string
-  options: ClientOptions
-  adminKey: bsv.HDPrivateKey | null
-  logger: Logger
- 
+  logger: Logger;
+  http: Requester;
+
+  private xPriv?: bsv.HDPrivateKey;
 
   constructor(serverUrl: string, options: ClientOptions, loggerConfig: LoggerConfig = defaultLogger) {
-    this.serverUrl = serverUrl
-    this.adminKey = null
-    this.logger = makeLogger(loggerConfig)
-
-    if (options.xPriv) {
-      options.xPrivString = options.xPriv.toString()
-      options.xPub = options.xPriv.hdPublicKey
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPrivString) {
-      options.xPriv = bsv.HDPrivateKey.fromString(options.xPrivString)
-      options.xPub = options.xPriv?.hdPublicKey
-      options.xPubString = options.xPub?.toString()
-    } else if (options.xPub) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPubString = options.xPub.toString()
-    } else if (options.xPubString) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = bsv.HDPublicKey.fromString(options.xPubString)
-    } else if (options.accessKey) {
-      options.xPriv = undefined
-      options.xPrivString = undefined
-      options.xPub = undefined
-      options.xPubString = undefined
-      options.accessKeyString = options.accessKey.toString()
+    this.logger = makeLogger(loggerConfig);
+
+    if (options.signRequest && options.xPriv) {
```

Comment:
Why this `if` statement is outside the initRequester?

Reply:
In current version it is no longer there.